packages/core/src/getStateFromPath.tsx

```diff
@@ -159,7 +157,46 @@ export default function getStateFromPath(
       remaining = segments.join('/');
     }
 
-    const state = createNestedStateObject(routeNames, initialRoutes, params);
+    const state = createNestedStateObject(
+      routeNames.map((name) => {
```

Comment:
Maybe extract this sometime. Kind of big block of code that is a param of a function.